src/v/cloud_storage/remote_partition.cc

```diff
@@ -81,7 +81,21 @@ class partition_record_batch_reader_impl final
         }
     }
 
-    ~partition_record_batch_reader_impl() override = default;
+    ~partition_record_batch_reader_impl() override {
```

Comment:
should this be made noexcept? returning the reader looks like it interacts with quite a few things that might throw--memory allocations, a semaphore that could be broken. are we prepared for that?

Reply:
I've updated this to eat+log exceptions in this destructor.

In the case of something like a bad-alloc when appending the reader to its segment's list, we'll end up trying to destroy the reader there&then.  That can be fatal in itself (remote_segment_batch_reader asserts that its stop() was called) but since it's entirely possible that stop() was called before this destructor, we permit it and log the error.